dispatcher/dispatcher.py

```diff
@@ -29,29 +29,35 @@ def main():
             default='config.ini')
     parser.add_argument('--log', type=str, help='Logging level', default='DEBUG',
             choices=['DEBUG','INFO','WARNING','ERROR','CRITICAL'])
+    parser.add_argument('--test', action=store_true, help='Are you testing?')
     args = parser.parse_args()
     config = configparser.ConfigParser()
     config.read(args.config)
-    logger = get_daq_logger('dispatcher', level = getattr(logging, args.log))
-    # Declare database object
-    MongoConnector = MongoConnect(config, logger)
-
-    # Declare a 'brain' object. This will cache info about the DAQ state in order to
-    # solve the want/have decisions. The deciding functions also go here. It needs a
-    # mongo connector because it has to pull options files
-    DAQControl = DAQController(config, MongoConnector, logger)
-    sleep_period = int(config['DEFAULT']['PollFrequency'])
+    config = config['DEFAULT' if not args.test else "TESTING"]
+    config['MasterDAQConfig'] = json.loads(config['MasterDAQConfig'])
+    control_uri = config['ControlDatabaseURI']%os.environ['MONGO_PASSWORD']
+    control_mc = pymongo.MongoClient(control_uri)
+    runs_uri = config['RunsDatabaseURI']%os.environ['RUNS_MONGO_PASSWORD']
+    runs_mc = pymongo.MongoClient(runs_uri)
+    logger = daqnt.get_daq_logger(config['LogName'], level=args.log, mc=control_mc)
+
+    # Declare necessary classes
     sh = SignalHandler()
+    Hypervisor = daqnt.Hypervisor(control_mc[config['ControlDatabaseName']], logger, sh)
```

Comment:
ok, now I see what this mysterious SH is

Reply:
and where it gets all initialized

Reply:
Especially when there are multiple threads, the signal handler allows you to shut down cleanly and not resort to kill-9, because kill-15 usually only hits the main thread.